### PR TITLE
Apply filter logic to disabled jobs dynamically

### PR DESCRIPTION
**Pull request by huydhn**
Apply filter logic to disabled jobs dynamically.  The list of disabled jobs is published at https://ossci-metrics.s3.amazonaws.com/disabled-jobs.json.  When the workflow (i.e. `pull`) and the platform (i.e. `linux-bionic-py3.8-clang9`) names match, job will be disabled (skipped) if they are in the list.

Note that getting the current job name within the GitHub action is fairly hacky.  This is a TODO item.

### Testing

* Unit testing
* This PR. https://github.com/pytorch/pytorch/issues/94861 disables `pull / linux-bionic-py3.8-clang9 / test (dynamo)` in the CI.  We have:
   * No dynamo tests running in `pull / linux-bionic-py3.8-clang9` https://github.com/pytorch/pytorch/actions/runs/4272505289/jobs/7437706181
   * Other dynamo tests, i.e. `pull / linux-bionic-py3.11-clang9`, are run normally https://github.com/pytorch/pytorch/actions/runs/4272505289/jobs/7437706054
 * This PR. https://github.com/pytorch/pytorch/issues/95642 disables `pull / linux-bionic-cuda11.7-py3.10-gcc7-sm86 / test`.  All test jobs for `pull / linux-bionic-cuda11.7-py3.10-gcc7-sm86` are skipped https://github.com/pytorch/pytorch/actions/runs/4287330986/jobs/7468179694
 * This PR. https://github.com/pytorch/pytorch/issues/95656 disables `pull / linux-bionic-py3_8-clang8-xla / build`.  All build and test jobs for `pull / linux-bionic-py3_8-clang8-xla` are skipped https://github.com/pytorch/pytorch/actions/runs/4287330986/jobs/7470478905

### CODE REVIEW

File: .github/scripts/filter_test_configs.py [hunk 1]
```diff
@@ -1,56 +1,89 @@
...
-    "inductor_torchbench",
-    "jit_legacy",
-    "multigpu",
-    "nogpu_AVX512",
-    "nogpu_NO_AVX2",
-    "slow",
-    "tsan",
-    "xla",
-}}
+VALID_TEST_CONFIG_LABELS = {
```
Comment (huydhn) [inline .github/scripts/filter_test_configs.py:17]: Lintrunner ufmt change.  I have an action item to remove this hardcoded list later.

**Comment by huydhn** [comment]
Follow GitHub suggestion to dump the whole GitHub context for debugging https://github.com/pytorch/pytorch/actions/runs/4295649226/jobs/7486369571, only to confirm that the job name is definitely not there.  For example, https://paste.sh/azEaSu-Q#HcKWBRnczcNgQ9U9kplYHQlk only show job name as `build` instead of the full name `linux-bionic-py3.8-clang9 / build`

**Review by clee2000 — APPROVED** [review]
I'm not a big fan of how the build job looks like it ran but was actually skipped, but that kind of change would probably require another filtering step before all the build jobs start

**Comment by huydhn** [comment]
> I'm not a big fan of how the build job looks like it ran but was actually skipped, but that kind of change would probably require another filtering step before all the build jobs start

Yeah, that's the only reason I keep filter step as part of build, trying to chose between 2 not so good choices 1) green but skipped or 2) adding a new filter step before build.  My assumption is that it's rare to disable/skip a build job, thus the current approach we notice, i.e. picking the less intrusive change

**Comment by huydhn** [comment]
Note to myself:  To close all test disabled issues and make sure that the JSON file is empty before merging this one.

**Pull request closed without merging**